MapboxNavigation/RouteMapViewController.swift

```diff
@@ -644,110 +572,234 @@ extension RouteMapViewController: NavigationViewDelegate {
+        NotificationCenter.default.removeObserver(self, name: UIApplication.didEnterBackgroundNotification, object: nil)
+        NotificationCenter.default.removeObserver(self, name: .routeControllerDidPassVisualInstructionPoint, object: nil)
+    }
+
+    func showStatus(title: String, withSpinner spin: Bool = false, for time: TimeInterval, animated: Bool = true, interactive: Bool = false) {
+        self.statusView.show(title, showSpinner: spin, interactive: interactive)
+        guard time < .infinity else { return }
+        self.statusView.hide(delay: time, animated: animated)
+    }
+
+    func setCamera(altitude: Double) {
+        guard self.mapView.altitude != altitude else { return }
+        self.mapView.altitude = altitude
+    }
+
+    func populateName(for waypoint: Waypoint, populated: @escaping (Waypoint) -> Void) {
+        guard waypoint.name == nil else { return populated(waypoint) }
+        CLGeocoder().reverseGeocodeLocation(waypoint.location) { places, error in
+            guard let place = places?.first, let placeName = place.name, error == nil else { return }
+            let named = Waypoint(coordinate: waypoint.coordinate, name: placeName)
+            return populated(named)
+        }
+    }
+	
+    func removePreviewInstructions() {
+        if let view = previewInstructionsView {
+            view.removeFromSuperview()
+            self.navigationView.instructionsBannerContentView.backgroundColor = InstructionsBannerView.appearance().backgroundColor
+            self.navigationView.instructionsBannerView.delegate = self
+            self.previewInstructionsView = nil
+        }
+    }
+	
+    func displayPreviewInstructions() {
+        self.removePreviewInstructions()
+
+        if let controller = stepsViewController {
+            self.stepsViewController = nil
+            controller.dismiss()
+        } else if let routeController {
+            let controller = StepsViewController(routeProgress: routeController.routeProgress)
+            controller.delegate = self
+            addChild(controller)
+            view.insertSubview(controller.view, belowSubview: self.navigationView.instructionsBannerContentView)
+
+            controller.view.topAnchor.constraint(equalTo: self.navigationView.instructionsBannerContentView.bottomAnchor).isActive = true
+            controller.view.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+            controller.view.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+            controller.view.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
+
+            controller.didMove(toParent: self)
+            controller.dropDownAnimation()
+
+            self.stepsViewController = controller
+            return
+        }
+    }
+	
+    func roadFeature(for line: MLNPolylineFeature) -> (roadName: String?, shieldName: NSAttributedString?) {
```

Comment:
Is there a reason this method is no longer private?

Reply:
its still private, just in private extension